adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m

```diff
@@ -0,0 +1,183 @@
+    [completionHandler copy];
+    _interstitialRenderCompletionHandler = ^id<GADMediationInterstitialAdEventDelegate>(id<GADMediationInterstitialAd> interstitialAd, NSError *error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationInterstitialAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(interstitialAd, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    GADMAdapterInMobiInterstitialAd *__weak weakSelf = self;
+    NSString *accountID = _interstitialAdConfig.credentials.settings[GADMAdapterInMobiAccountID];
+    [GADMAdapterInMobiInitializer.sharedInstance
+     initializeWithAccountID:accountID
+     completionHandler:^(NSError *_Nullable error) {
+        GADMAdapterInMobiInterstitialAd *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return;
+        }
+        
+        if (error) {
+            NSLog(@"[InMobi] Initialization failed: %@", error.localizedDescription);
+            strongSelf->_interstitialRenderCompletionHandler(nil, error);
+            return;
+        }
+        
+        [strongSelf requestInterstitialAd];
+    }];
+}
+
+- (void)requestInterstitialAd {
+    long long placementId =
+    [_interstitialAdConfig.credentials.settings[GADMAdapterInMobiPlacementID] longLongValue];
+    if (placementId == 0) {
+        NSError *error = GADMAdapterInMobiErrorWithCodeAndDescription(
+                                                                      GADMAdapterInMobiErrorInvalidServerParameters,
+                                                                      @"[InMobi] Error - Placement ID not specified.");
+        _interstitialRenderCompletionHandler(nil,error);
+        return;
+    }
+    
+    if ([_interstitialAdConfig isTestRequest]) {
+        NSLog(@"[InMobi] Please enter your device ID in the InMobi console to receive test ads from "
+              @"InMobi");
+    }
+    
+    _interstitialAd = [[IMInterstitial alloc] initWithPlacementId:placementId];
+    
+    GADInMobiExtras *extras =  _interstitialAdConfig.extras;
+    if (extras && extras.keywords) {
+        [_interstitialAd setKeywords:extras.keywords];
+    }
+    
+    GADMAdapterInMobiSetTargetingFromAdConfiguration(_interstitialAdConfig);
+    NSDictionary<NSString *, id> *requestParameters =
+    GADMAdapterInMobiCreateRequestParametersFromAdConfiguration(_interstitialAdConfig);
+    [_interstitialAd setExtras:requestParameters];
+    
+    _interstitialAd.delegate = self;
+    [_interstitialAd load];
+}
+
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    if ([_interstitialAd isReady]) {
+        [_interstitialAd showFromViewController:viewController
+                                  withAnimation:kIMInterstitialAnimationTypeCoverVertical];
+    }
```

Comment:
We should add an `else` statement and return a failure callback, like what you did in rewarded ad.